File: app/src/main/java/it/paranoidsquirrels/idleguildmaster/storage/data/entities/enemies/units/AvatarOfTheAncient.java
package it.paranoidsquirrels.idleguildmaster.storage.data.entities.enemies.units;

import it.paranoidsquirrels.idleguildmaster.R;
import it.paranoidsquirrels.idleguildmaster.storage.data.entities.Skills;
import it.paranoidsquirrels.idleguildmaster.storage.data.entities.enemies.Enemy;
import it.paranoidsquirrels.idleguildmaster.storage.data.items.ItemWrapper;
import java.util.LinkedHashMap;

/* JADX INFO: loaded from: classes3.dex */
public class AvatarOfTheAncient extends Enemy {
    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.entities.enemies.Enemy
    protected int getMaxDamage() {
        return 650;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.entities.enemies.Enemy
    protected int getMinDamage() {
        return 610;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.entities.Entity
    public boolean isMagic() {
        return true;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.entities.Entity
    public boolean isRanged() {
        return false;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.entities.enemies.Enemy
    protected void configureStatistics() {
        this.baseMaxHp = 6000;
        this.baseConstitution = 1;
        this.baseIntelligence = 999;
        this.baseDexterity = 24;
        this.baseDefense = 0;
        this.baseMagicDefense = 0;
        this.flatDodgeChance = 0.6d;
        this.immunityToStatus = 1.0d;
        this.imageId = R.drawable.unit_avatar_of_the_ancient;
        this.idName = R.string.enemy_avatar_of_the_ancient_name;
        this.idDescription = R.string.enemy_avatar_of_the_ancient_description;
        this.passiveSkill = Skills.PASSIVE_COSMIC_PROJECTION;
        this.activeSkill = Skills.ACTIVE_NONE;
        this.rarity = 1;
        this.expGiven = 10000;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.entities.enemies.Enemy
    public LinkedHashMap<ItemWrapper, Integer> listDrops(int i) {
        LinkedHashMap<ItemWrapper, Integer> linkedHashMap = new LinkedHashMap<>();
        linkedHashMap.put(ItemWrapper.getInstance("StarFragment", 1), 325);
        linkedHashMap.put(ItemWrapper.getInstance("EsotericEgg", 1), 1);
        return linkedHashMap;
    }
}

File: app/src/main/kotlin/it/paranoidsquirrels/idleguildmaster/storage/data/entities/enemies/units/AvatarOfTheAncient.kt
package it.paranoidsquirrels.idleguildmaster.storage.data.entities.enemies.units

import it.paranoidsquirrels.idleguildmaster.R
import it.paranoidsquirrels.idleguildmaster.storage.data.entities.Skills
import it.paranoidsquirrels.idleguildmaster.storage.data.entities.enemies.Enemy
import it.paranoidsquirrels.idleguildmaster.storage.data.items.ItemWrapper
import java.util.LinkedHashMap

class AvatarOfTheAncient : Enemy() {
    override fun getMaxDamage(): Int = 650
    override fun getMinDamage(): Int = 610
    override fun isMagic(): Boolean = true
    override fun isRanged(): Boolean = false

    override fun configureStatistics() {
        baseMaxHp = 6000
        baseConstitution = 1
        baseIntelligence = 999
        baseDexterity = 24
        baseDefense = 0
        baseMagicDefense = 0
        flatDodgeChance = 0.6
        immunityToStatus = 1.0
        imageId = R.drawable.unit_avatar_of_the_ancient
        idName = R.string.enemy_avatar_of_the_ancient_name
        idDescription = R.string.enemy_avatar_of_the_ancient_description
        passiveSkill = Skills.PASSIVE_COSMIC_PROJECTION
        activeSkill = Skills.ACTIVE_NONE
        rarity = 1
        expGiven = 10000
    }

    override fun listDrops(i: Int): LinkedHashMap<ItemWrapper, Int> {
        val linkedHashMap = LinkedHashMap<ItemWrapper, Int>()
        linkedHashMap.put(ItemWrapper.getInstance("StarFragment", 1), 325)
        linkedHashMap.put(ItemWrapper.getInstance("EsotericEgg", 1), 1)
        return linkedHashMap
    }
}
